flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -207,6 +208,9 @@
 	/** Unique ID of this backend. */
 	private UUID backendUID;
 
+	/** The byte array for namespace serialization in getKeys(). */
+	private final ByteArrayOutputStreamWithPos namespaceOutputStream;
```

Comment:
It feels like this member is in a too broad scope. While this maximizes caching, I wonder if creating one of there per `getKeys(...)` all is not cleaner and still efficient enough. What do you think?

Reply:
I would choose to create one of there per` getKeys(...)`.. I think it is cleaner to me than put it into `RocksDBKeyedStateBackend`, cause it is only used in `getKeys()` for now.

Reply:
Ok, sounds good 👍